### ISSUE TITLE
Using EmitAsync with CancellationToken corrupts the request

### ISSUE THREAD

**Issue by Jakobi-mirsk** [doghappy/socket.io-client-csharp#171]
hi
using this method to send a sting, corrupts the received message with strange structure.

public Task EmitAsync(string eventName, CancellationToken cancellationToken, params object[] data)

**Comment by doghappy** [comment]
Hi, can you describe in detail?

**Comment by Jakobi-mirsk** [comment]
we had a setup where the client was receiving what ever it send to the server. So when the client send 'Hello' to the server, the server would send it back to the client. I the client used EmitAsync(string eventName, params object[] data), it would get back the 'Hello' in the message, but when using it with the CancellationToken we get a structure back that looks like something from the token structure(i have not checked if it is, but it looks like a C# structure parsed to json)

**Comment by doghappy** [comment]
This is my test code:

**server**

```js
    socket.on('hi', (msg) => {
        socket.emit('hi', "V3" + msg);
    });
```

**client**

```cs
await socket.EmitAsync("hi", "test1: " + DateTime.Now.ToString());
await socket.EmitAsync("hi", CancellationToken.None, "test2: " + DateTime.Now.ToString());
```

**output**

```
server: V3: test1: 2021/7/14 15:18:43
server: V3: test2: 2021/7/14 15:18:43
```

I failed to reproduce your issue. Do I need to modify my test code?

**Comment by jlareo** [comment]
Could it be that you are providing the wrong order of the parameters?:

`await socket.EmitAsync("hi", "test", **CancellationToken.None**);`

Instead of:

`await socket.EmitAsync("hi", **CancellationToken.None**, "test");`

In that case CancellationToken will be detected as an element of params an then will be serialized as json.

**Comment by Jakobi-mirsk** [comment]
I am coding in C# so wrong order would throw a compile error.

**Comment by jlareo** [comment]
As I explained will be detected as an element of params so you will not get an exception.

**Comment by Jakobi-mirsk** [comment]
Exactly. Sorry I did not understand your explanation when I read it.

**Comment by jlareo** [comment]
Generally it is a good practise to put the CancellationToken at the last position, but unfortunatelly by having params is not possible.

**Comment by Jakobi-mirsk** [comment]
Okay, i thought you meant something else. Problem not solved :D I was calling it like this:
`await socket.EmitAsync("hi", CancellationToken.None, "test");`
and still got this result:
`{"IsCancellationRequested":false,"CanBeCanceled":false,"WaitHandle":{"Handle":{},"SafeWaitHandle":{"IsInvalid":false,"IsClosed":false}}}`

So event though mi call the method correctly it still thinks its a parameter. So basically that version of the method does not work in C#, as i see it.

**Comment by jlareo** [comment]
You could try to use the name of he parameter before the value:

await socket.EmitAsync("hi", cancellationToken: CancellationToken.None, "test");

But I don't understand how it is possible that for me it was detected as a CancellationToken but it is not for you.

**Comment by Jakobi-mirsk** [comment]
yes, strange. could it be .NET version. I am using 4.6.2

**Comment by doghappy** [comment]
Has this issue been resolved?
